client/v2/autocli/flag/bool_to_value.go

```diff
@@ -0,0 +1,137 @@
+package flag
+
+import (
+	"github.com/spf13/pflag"
+	"strconv"
+)
+
+func newBoolToString[K bool, V string](val map[K]V, p *map[K]V) *genericMapValue[K, V] {
+	newBoolStringMap := newGenericMapValue(val, p)
+	newBoolStringMap.Options = genericMapValueOptions[K, V]{
+		genericType: "boolToString",
+		keyParser: func(s string) (K, error) {
+			value, err := strconv.ParseBool(s)
+			return K(value), err
```

Comment:
Unnecessary, please just return `return strconv.ParseBool(s)` or even better
```go
keyParser: strconv.ParseBool,
```

Reply:
Sorry @odeke-em could you provide me with some guidance on how to handle this error? 
`Cannot use 'strconv.ParseBool' (type func(str string) (bool, error)) as the type func(string) (K, error)`
In order to make this change